src/Monitor/Monitor/ScheduledQueryRules/NewScheduledQueryRuleSourceCommand.cs

```diff
@@ -0,0 +1,54 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Insights.OutputClasses;
+using Microsoft.Azure.Management.Monitor.Models;
+using System.Collections.Generic;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Insights.ScheduledQueryRules
+{
+    /// <summary>
+    /// Create a ScheduledQueryRule Source object
+    /// </summary>
+    [Cmdlet(VerbsCommon.New, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "ScheduledQueryRuleSource"), OutputType(typeof(PSScheduledQueryRuleSource))]
+    public class NewScheduledQueryRuleSourceCommand : ManagementCmdletBase
+    {
+
+        #region Cmdlet parameters
+
+        [Parameter(Mandatory = true, HelpMessage = "The alert query")]
+        [ValidateNotNullOrEmpty]
+        public string Query { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "The list of authorized resources for this alert")]
+        public List<string> AuthorizedResource { get; set; }
```

Comment:
@shpimpal same comment about changing this to `string[]`

Reply:
@shpimpal ping on this comment

Reply:
addressed